### DISCUSSION TITLE
Could not find a version that satisfies the requirement textual (from versions: )

### DISCUSSION THREAD

**Discussion by StolleJay** [Textualize/textual#4494] [Q&A]
I try to install textual on my raspberry pi b+ and get with the `pip install textual` or `sudo pip install textual` the following error message:
```
Could not find a version that satisfies the requirement textual (from versions: )
No matching distribution found for textual
```

If i try it with `pip3 install textual` it works but then after creating my first simple app (just a window) it says:
```
from textual.app import App
ModuleNotFoundError: No module named 'textual'
```

What do i wrong?
I just want to create a nice GUI with css.
TkInter what i use currently has not really "modern" optics...

OS: **Raspbian GNU/Linux 10 (buster) Linux 5.10.103+**
Python: **2.7.16**
Python 3: **3.7.3**

**Comment by davep** [comment]
I would imagine one major cause of problems would be that Textual only supports currently-supported versions of Python; so Python 3.8 or above.

**Comment by StolleJay** [comment]
I tried it now on my pi zero 2 w with `python3 3.9.2` on **Raspberry Pi OS 32bit (Legacy)** and there i get the same message:
```
from textual.app import App
ModuleNotFoundError: No module named 'textual'
```

This is my app from the *App Basics*:
```
from textual.app import App


class MyApp(App):
    pass


if __name__ == "__main__":
    app = MyApp()
    app.run()
```

**Reply by davep** [reply]
Switching to Python 3.9 will have solved this problem:

> Could not find a version that satisfies the requirement textual (from versions: )
> No matching distribution found for textual

The problem you now have would suggest you're installing Textual in an environment that isn't the environment used when you run that application. It's tricky to diagnose that as the cause will depend on how you've set up and installed Python and any virtual environments, etc.

I would expect you to run into the same issue with any non-core Python library, not just Textual.
